### PR TITLE
Add bit-manipulation concept

### PR DESCRIPTION
**Pull request by kahgoh**
Following on from #552, a draft of the secrets exercise has been added for the bit manipulation concept.

### CODE REVIEW

File: concepts/bit-manipulation/about.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Bit Manipulation
+
+Crystal has bitwise operators for manipulating [`Integers`][integers] at the binary level.
```
Comment (meatball133) [inline concepts/bit-manipulation/about.md:3]: The class for integers in Crystal is called `Int`

File: concepts/bit-manipulation/links.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "url": "https://crystal-lang.org/reference/1.10/syntax_and_semantics/operators.html#binary",
```
Comment (meatball133) [inline concepts/bit-manipulation/links.json:3]: I thinked you missed making this not version specfic docs, e.g "https://crystal-lang.org/reference/syntax_and_semantics/operators.html#binary"

File: exercises/concept/secrets/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Your has just sent you a message with an important secret.
+Not wanting to make it easy for others to read it, the message has been encrypted using a series of bit manipulations.
+You will need to write a program to help decrypt the message.
+
+## 1. Shift back the bits
+
+The first step in decrypting the message is to undo the shifting from the encryption process by shifting the bits back to the left.
+The number of places to shift can vary between messages.
+
+Implement the method `Secrets.shift_back` that takes a value and the number of places to shift left.
```
Comment (meatball133) [inline exercises/concept/secrets/.docs/instructions.md:12]: This doesnt match how the method is tested in the spec files. Should be `Secrets::shift_back`

Comment (kahgoh) [reply exercises/concept/secrets/.docs/instructions.md:12]: With this one, I ended deciding to update the tests, stub and exemplar files to match.

File: exercises/concept/secrets/src/secrets.cr [hunk 4]
```diff
@@ -1,17 +1,17 @@
 module Secrets
-  def shift_back(value : UInt8, numberOfPlaces : UInt8) : UInt8
+  def self.shift_back(value : UInt8, numberOfPlaces : UInt8) : UInt8
     raise "Please implement the Secrets::shift_back method"
```
Comment (meatball133) [inline exercises/concept/secrets/src/secrets.cr:3]: This text has to be updated to reflect the changes

**Review by meatball133 — APPROVED** [review]

**Pull request merged**